pkg/controller/node/node_controller.go

```diff
@@ -183,14 +182,14 @@ func (ctrl *Controller) checkMasterNodesOnDelete(obj interface{}) {
 	}
 	currentMasters, err := ctrl.getCurrentMasters()
 	if err != nil {
-		goerrs.Wrap(err, "Reconciling to make master nodes schedulable/unschedulable failed")
+		fmt.Errorf("Reconciling to make master nodes schedulable/unschedulable failed: %w", err)
```

Comment:
golangci-lint caught this. The reason why this was never caught before is most likely because the linter was not looking for the `goerrs.Wrap()` function. When this was changed to use `fmt.Errorf()`, the linter picked up on the pattern and alerted to an unused result since `fmt.Errorf()` is likely a more well-known signature.

That explanation aside, it's interesting that this error is never bubbled out of this function. This could be because this function is passed into a [ResourceEventHandlerFuncs](https://github.com/kubernetes/client-go/blob/master/tools/cache/controller.go#L218-L226) struct whose type does not accept a function with any return arguments. The same pattern can be seen throughout `ctrl.checkMasterNodesOnDelete`, `ctrl.reconcileMaster` and `ctrl.reconcileMasters` (all within this file, `pkg/controller/node/node_controller.go`).

As for a proposed remedy, I offer two suggestions:
1. We replace these specific calls to `fmt.Errorf()` with calls to `glog.Errorf()` so the errors are logged. This remedy would not introduce any unknown behavior and may help gain insight into how often these particular errors occur.
2. Instead of just logging the error, we handle it using the [`utilruntime.HandleError()`](https://github.com/kubernetes/apimachinery/blob/master/pkg/util/runtime/runtime.go#L83-L111) function which will try again using the `rudimentaryErrorBackoff` mechanism within that package. This function appears to be used throughout this file. However, this could introduce unknown behavior.

Reply:
Interesting. I think paths like this one are very rarely considered in the first place, since we don't generally consider node deletion operations. In fact I am not sure whether we have any testing for this case, and whether we'd be able to auto recover, if at all.

I think right now I am leaning towards option 1, since I think it's fine to increase error logging in general in case these situations do happen. For 2, I don't know how we are handling the errors at all, so I am not sure what behaviour the HandleError actually exhibits for our use case (again, they should very rarely fire)

Reply:
Should we leave option 2 for now? I have made a commit for option 1. I can go ahead and squash it with the previous commits (after a rebase since there are merge conflicts).